crates/exec-wasmtime/src/runtime/vfs/listen.rs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! A file system containing listening networking sockets.
+
+use super::super::net::tls;
+use super::super::WasiResult;
+
+use std::any::Any;
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+use std::ops::Deref;
+use std::path::MAIN_SEPARATOR;
+use std::sync::Arc;
+
+use anyhow::Context;
+use cap_std::net::TcpListener;
+use enarx_config::{FileName, ListenFile};
+use rustls::{Certificate, PrivateKey};
+use wasi_common::file::{FdFlags, FileType};
+use wasi_common::{Error, ErrorExt, WasiDir, WasiFile};
+use wasmtime_vfs_dir::Directory;
+use wasmtime_vfs_ledger::InodeId;
+use wasmtime_vfs_memory::{Data, Inode, Link, Node};
+use wiggle::async_trait;
+use zeroize::Zeroizing;
+
+enum Socket {
+    Tcp(Link<()>),
+    Tls(Link<Arc<rustls::ServerConfig>>),
+}
+
+#[async_trait]
```

Comment:
If you choose to use async traits above, use it here as well.

Reply:
no